core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1997,6 +2003,11 @@ private CompiledQuery insert(ExecutionModel executionModel, SqlExecutionContext
                 RecordToRowCopier copier = RecordToRowCopierUtils.generateCopier(asm, record, metadata, listColumnFilter);
                 insertOperation.addInsertRow(new InsertRowImpl(record, copier, timestampFunction, tupleIndex));
             }
+
+            if (hasBindVariable) {//we need column names only for inserts with bind vars that are cached
```

Comment:
"we need column names only for inserts with bind vars that are cached" -> what about when it is an insert without bind variables but needs to rerun checks because user name and/or version is not matching? why do not we need the columns in that case?

Reply:
Insert statements without any bind variables aren't cached so their factories are not reused and don't need re-authentication . Have a look at PGConnectionContext.java :
```java
case CompiledQuery.INSERT:
                queryTag = TAG_INSERT;
                typesAndInsert = typesAndInsertPool.pop();
                typesAndInsert.of(cq.getInsertOperation(), bindVariableService);
                if (bindVariableService.getIndexedVariableCount() > 0) {
                    LOG.debug().$("cache insert [sql=").$(queryText).$(", thread=").$(Thread.currentThread().getId()).$(']').$();
                    // we can add insert to cache right away because it is local to the connection
                    typesAndInsertCache.put(queryText, typesAndInsert);
                }
```

Reply:
got it, thanks.
actually it makes sense to cache inserts only with bind variables, unlikely that user wants to insert dupes into the table.

Reply:
Turns out that after rework it is possible to hit the scenario when if access list is refreshed between insert compilation and execution so I fixed it in lastest commit .